### PR TITLE
Reintroduce MediaTrackSource

### PR DESCRIPTION
**Pull request by fibann**
Now that there is no VideoRendererSource we don't need it to be an interface anymore.

### CODE REVIEW

File: libs/unity/library/Runtime/Scripts/Media/MediaTrackSource.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+        private readonly List<MediaLine> _mediaLines = new List<MediaLine>();
+
+        internal void OnAddedToMediaLine(MediaLine mediaLine)
+        {
+            Debug.Assert(!_mediaLines.Contains(mediaLine));
+            _mediaLines.Add(mediaLine);
+        }
+
+        internal void OnRemoveFromMediaLine(MediaLine mediaLine)
```
Comment (djee-ms) [inline libs/unity/library/Runtime/Scripts/Media/MediaTrackSource.cs:34]: nit: for symmetry
```suggestion
        internal void OnRemovedFromMediaLine(MediaLine mediaLine)
```

File: libs/unity/library/Runtime/Scripts/Media/MediaLine.cs [hunk 2]
```diff
@@ -51,40 +51,35 @@ public class MediaLine : ISerializationCallbackReceiver
...
+                    if (_source != value)
                     {
-                        if (_source != mediaTrackSource)
+                        if (_source != null)
                         {
-                            (_source as IMediaTrackSourceInternal)?.OnRemoveFromMediaLine(this);
-                            _source = mediaTrackSource;
-                            (_source as IMediaTrackSourceInternal).OnAddedToMediaLine(this);
-                            CreateSenderIfNeeded();
+                            _source.OnRemoveFromMediaLine(this);
```
Comment (djee-ms) [inline libs/unity/library/Runtime/Scripts/Media/MediaLine.cs:78]: nit: `_source?.OnRemoveFromMediaLine(this);` without the explicit `if`?

Comment (fibann) [reply libs/unity/library/Runtime/Scripts/Media/MediaLine.cs:78]: `_source` is now a MonoBehaviour and `?.` will give a warning

Comment (djee-ms) [reply libs/unity/library/Runtime/Scripts/Media/MediaLine.cs:78]: Right, nevermind.

**Pull request merged**